src/orders/components/OrderMetadataDialog/OrderMetadataDialog.tsx

```diff
@@ -73,80 +38,29 @@ export const OrderMetadataDialog = ({ onClose, open, order }: OrderMetadataDialo
 
   const { control, getValues, formState, trigger, reset } = formMethods;
 
-  // Metadata field arrays
-  const metadataControls = useFieldArray<MetadataFormData>({
+  const {
+    metadataFields,
+    privateMetadataFields,
+    handleMetadataChange,
+    handlePrivateMetadataChange,
+    metadataErrors,
+    privateMetadataErrors,
+  } = useMetadataFormControls({
```

Comment:
why dont you pull useForm to this hook? Now you initialize form outside and put it here. I think you can just put everythging inside this hook. useForm above is the same for every other modal